File: inject-java/src/test/groovy/io/micronaut/aop/introduction/NotImplementedAdvice.java
package io.micronaut.aop.introduction;

import io.micronaut.aop.MethodInterceptor;
import io.micronaut.aop.MethodInvocationContext;

import javax.inject.Singleton;

@Singleton
public class NotImplementedAdvice implements MethodInterceptor<Object, Object> {
    public static boolean invoked = false;
    @Override
    public Object intercept(MethodInvocationContext<Object, Object> context) {
        invoked = true;
        return context.proceed();
    }
}

File: inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/aop/introduction/NotImplementedAdvice.kt
package io.micronaut.kotlin.processing.aop.introduction

import io.micronaut.aop.MethodInterceptor
import io.micronaut.aop.MethodInvocationContext
import jakarta.inject.Singleton

@Singleton
class NotImplementedAdvice : MethodInterceptor<Any, Any> {
    var invoked = false

    override fun intercept(context: MethodInvocationContext<Any, Any>): Any? {
        invoked = true
        return context.proceed()
    }
}
